lib/spoom/location.rb

```diff
@@ -15,42 +15,77 @@ class << self
       sig { params(location_string: String).returns(Location) }
       def from_string(location_string)
         file, rest = location_string.split(":", 2)
-        raise LocationError, "Invalid location string: #{location_string}" unless file && rest
+        raise LocationError, "Invalid location. File name is missing in location string: #{location_string}" unless file
+
+        return new(file) if rest.nil?
 
         start_line, rest = rest.split(":", 2)
-        raise LocationError, "Invalid location string: #{location_string}" unless start_line && rest
+        if rest.nil?
+          start_line, end_line = T.must(start_line).split("-", 2)
+          return new(file, start_line: start_line.to_i, end_line: end_line.to_i) if end_line
+
+          raise LocationError, "Invalid location. End line is missing in range format: #{location_string}"
```

Comment:
I'd prefer an early raise and an unconditional return:

```suggestion
          start_line, end_line = T.must(start_line).split("-", 2)
          raise LocationError, "Invalid location. End line is missing in range format: #{location_string}" unless end_line
          
          return new(file, start_line: start_line.to_i, end_line: end_line.to_i) if end_line
```

So the style and logic is coherent with the other branches.